### PR TITLE
Refactor admin permissions endpoint to accept array of permissions

### PR DESCRIPTION
**Pull request by holabayor**
Refactored admin permissions endpoint to accept array of permissions and return improved response for the frontend

### CODE REVIEW

File: src/modules/admin/controllers/admin_staff.controller.ts [hunk 1]
```diff
@@ -248,36 +249,35 @@ export const addPermissionToStaff = async (
...
-                return res
-                    .status(401)
-                    .json({success: false, message: "Staff already has this permission" });
-            }
+        const existingPermissions = subAdmin.permissions.map((permission)=> permission.toString());
+        const newPermissions = Array.from(new Set([...existingPermissions, ...validPermissionIds]));
 
-            permissions.push(availabePermission)
+        // Check if the permission have changed
+        if (existingPermissions.length === newPermissions.length) {
```
Comment (AironDev) [inline src/modules/admin/controllers/admin_staff.controller.ts:270]: This is not needed

File: src/modules/admin/controllers/admin_staff.controller.ts [hunk 2]
```diff
@@ -248,36 +249,35 @@ export const addPermissionToStaff = async (
...
                 .json({ message: "staff does not exist" });
         }
 
-        const checkPermission = await PermissionModel.findOne({ _id: permision })
+        const validPermissions = await PermissionModel.find({ _id:{ $in: permissions } });
 
-        if (!checkPermission) {
+        if (validPermissions.length === 0) {
             return res
-                .status(401)
```
Comment (AironDev) [inline src/modules/admin/controllers/admin_staff.controller.ts:259]: Empty valid permissions can be used to clear all permissions from the staff.

**Review by AironDev — CHANGES_REQUESTED** [review]

File: src/modules/admin/controllers/admin_staff.controller.ts (hunk 1 above)
Comment (holabayor) [reply src/modules/admin/controllers/admin_staff.controller.ts:270]: That's the only way I'd know if a new permission was added or the same permissions are being added again.

File: src/modules/admin/controllers/admin_staff.controller.ts (hunk 2 above)
Comment (holabayor) [reply src/modules/admin/controllers/admin_staff.controller.ts:259]: line 256-262 will validate each permission and check if it's not empty

**Review by RepairFind-CA — APPROVED** [review]

**Pull request merged**